protocol/msgpack/unaggregated_iterator.go

```diff
@@ -166,12 +173,28 @@ func (it *unaggregatedIterator) decodeBatchTimer() {
 	}
 	it.metric.Type = unaggregated.BatchTimerType
 	it.metric.ID = it.decodeID()
-	numValues := it.decodeArrayLen()
-	values := it.floatsPool.Get(numValues)
+	var (
+		timerValues []float64
+		poolAlloc   = false
+		numValues   = it.decodeArrayLen()
+	)
+	if cap(it.timerValues) >= numValues {
+		it.timerValues = it.timerValues[:0]
+		timerValues = it.timerValues
+	} else if numValues <= it.largeFloatsSize {
+		it.timerValues = make([]float64, 0, numValues)
```

Comment:
Any reason why we wouldn't want to make the capacity of `it.timerValues` equal to `it.largeFloatSize` so we don't have to allocate here?